README.md

```diff
@@ -13,9 +13,32 @@ For runnable examples, see the [examples directory](examples/).
 
 BETA. Expect potential API changes.
 
+## Conatiner
+
+`package dig` exposes `type Container` as an object capable of resolving a
+directional dependency graph.
+
+To create one:
+```go
+import "go.uber.org/dig"
+
+func main() {
+	c := dig.New()
+	// dig container `c` is ready to use!
+}
+```
+
+Objects in the container are identified by their `reflect.Type` and **everything
+is treated as a singleton**, meaning there can be only one object in the graph
+of a given type.
+
+For more advanced usecases, consider using a factory pattern. That is to say,
```

Comment:
Yes, we want to hint that you can solve multi-types through OO, i.e. a factory. More fidelity can be added into it later